### PR TITLE
Remove zstandard dependency in favor of numcodecs

### PR DESCRIPTION
**Pull request by normanrz**
I added `zstandard` as a dependency because the `zstd` in numcodecs doesn't expose the `checksum` config. This will be introduced in https://github.com/zarr-developers/numcodecs/pull/519. Once that PR is merged, we can use the numcodecs version.

### CODE REVIEW

**Comment by mkitti** [comment]
The correct default level should be `0` which will then allow the Zstd C library to use the default setting according to how it was built:
https://github.com/facebook/zstd/blob/97291fc5020a8994019ab76cf0cda83a9824374c/lib/zstd.h#L338

File: src/zarr/codecs/zstd.py [hunk 1]
```diff
@@ -15,6 +16,8 @@
     from zarr.config import RuntimeConfiguration
     from zarr.common import BytesLike, JSON, ArraySpec
 
+DEFAULT_ZSTD_LEVEL = 3
```
Comment (mkitti) [inline src/zarr/codecs/zstd.py:19]:
```suggestion
DEFAULT_ZSTD_LEVEL = 0
```

Comment (mkitti) [reply src/zarr/codecs/zstd.py:19]: Here this should stay `0`. I'm not sure if we need this constant.

**Review by mkitti — CHANGES_REQUESTED** [review]
At this level the default should stay `0` which the Zstandard C library will interpret as use the default that was built with as defined by zstd.h:
https://github.com/facebook/zstd/blob/97291fc5020a8994019ab76cf0cda83a9824374c/lib/zstd.h#L338

**Comment by normanrz** [comment]
Now taht numcodecs 0.13 is out, we can move forward with this.

**Pull request merged**